apps/frontend/src/pages/tap.tsx

```diff
@@ -114,6 +118,36 @@ const TapPage: React.FC = () => {
             // Save tap to local storage
             await storage.addUserTap(response);
 
+            // If tap graph feature enabled, upsert graph edge (upsert so that either user can create the row)
+            let tapSenderHash: string | null = null;
+
+            const sentHash: boolean = (user?.tapGraphEnabled === true);
+            if (sentHash) {
+              // Double hash the signature private key, use as identifier, use single hash version as revocation code
+              tapSenderHash = sha256(sha256(user.signaturePrivateKey).concat(DEVCON));
+            }
+
+            // Upsert row, returns edge ID
+            try {
+              const resp: UpsertSocialGraphEdgeResponse = await upsertSocialGraphEdge(session.authTokenValue, null, tapSenderHash, null);
```

Comment:
why are we upserting with null tapSenderHashs?

Reply:
(1) because each side can only submit their own hash. 
(2) each side can choose to consenting to sending their hash. If they don't consent, we still create the edge so they can still add them later.